extensions/wikia/Privacy/UserDataRemover.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+use Wikia\Logger\Loggable;
+
+class UserDataRemover {
+	use Loggable;
+
+	public function removeGlobalData( int $userId ) {
+		try {
+			$user = User::newFromId( $userId );
+			if( $user->isAnon() ) {
+				return false;
+			}
+
+			$userIdentityBox = new UserIdentityBox( $user );
+			$userIdentityBox->clearMastheadContents();
+			Wikia::invalidateUser( $user, true, false );
```

Comment:
Let's move this cache invalidation section after the update + commit lines. Additionally, before the DB query let's just call `$user->invalidateCache` to make sure to invalidate old user name -> user ID mapping cache.

Reply:
Or even further: let's do what user rename process performs:

```php
		// SUS-3523: invalidate the destination user (by both name and ID, user name to ID
		// mapping is cached as well)
		$this->invalidateUser( User::newFromId( $this->mUserId ) );
		$this->invalidateUser( $this->mNewUsername );
```

Before and after an update query.